Fixed build error and AcquireToken function

sureshchahal opened the pull request:
Fixed acquireToken and build error

sureshchahal committed:
Fixed build error and acquire token function

sureshchahal committed:
allback when access token is received

sureshchahal committed:
Handled error during acquire token

sureshchahal committed:
Added Http extension for acquring token for CORS API

Neil665 commented:
Hi sureshchahal, thanks for adding this functionality! (Thanks also to alenny for the original code!)

I wanted to try out your code so I cloned it from https://github.com/sureshchahal/angular2-adal and then I ran 'npm install' and 'gulp build'. I got errors because there were two TypeScript declaration files for adal.js, so I removed the line in typings.json so I was only left with the one under 'src\adal'. However there were still a number of build errors, as shown below. Were you using a more updated declaration file for adal.js?

[16:39:17] Using gulpfile ~\git\sureshchahal\angular2-adal\gulpfile.js
[16:39:17] Starting 'build'...
[16:39:17] Starting 'clean:dist'...
[16:39:17] Finished 'clean:dist' after 9.92 ms
[16:39:17] Starting 'readme:dist'...
[16:39:17] Starting 'scripts:cjs'...
[16:39:17] Starting 'scripts:ts'...
[16:39:17] Finished 'readme:dist' after 260 ms
src\services\adal.service.ts(37,16): error TS2339: Property 'AuthenticationContext' does not exist on type 'Window'.
src\services\adal.service.ts(72,34): error TS2339: Property 'callback' does not exist on type 'AuthenticationContext'.
src\services\adal.service.ts(72,59): error TS2339: Property 'callBackMappedToRenewStates' does not exist on type 'Window'.
src\services\adal.service.ts(77,45): error TS2339: Property 'callback' does not exist on type 'AuthenticationContext'.
src\services\adal.service.ts(84,46): error TS2339: Property 'callback' does not exist on type 'AuthenticationContext'.
src\services\adal.service.ts(84,72): error TS2339: Property '_getItem' does not exist on type 'AuthenticationContext'.
src\services\adal.service.ts(84,98): error TS2339: Property 'CONSTANTS' does not exist on type 'AuthenticationContext'.
src\services\adal.service.ts(89,46): error TS2339: Property 'callback' does not exist on type 'AuthenticationContext'.
src\services\adal.service.ts(89,72): error TS2339: Property '_getItem' does not exist on type 'AuthenticationContext'.
src\services\adal.service.ts(89,98): error TS2339: Property 'CONSTANTS' does not exist on type 'AuthenticationContext'.
src\services\adal.service.ts(90,46): error TS2339: Property '_renewFailed' does not exist on type 'AuthenticationContext'.
src\services\authHttp.service.ts(2,39): error TS2307: Cannot find module '@angular/http'.
src\services\authHttp.service.ts(3,27): error TS2307: Cannot find module 'angular2-adal/core'.
[16:39:21] TypeScript: 13 semantic errors
[16:39:21] TypeScript: emit succeeded (with errors)
[16:39:21] Finished 'scripts:ts' after 4.63 s
[16:39:21] Finished 'scripts:cjs' after 4.72 s
[16:39:21] Starting 'scripts'...
[16:39:21] Finished 'scripts' after 4.03 μs
[16:39:21] Starting 'bundle:cjs'...
[16:39:23] Starting 'bundle:cjs:min'...
[16:39:45] Finished 'bundle:cjs' after 24 s
[16:39:45] Finished 'bundle:cjs:min' after 23 s
[16:39:45] Starting 'bundle'...
[16:39:45] Finished 'bundle' after 5.87 μs
[16:39:45] Starting 'copy-typings'...
[16:39:45] Finished 'copy-typings' after 2.92 ms
[16:39:45] Starting 'create-package-json'...
[16:39:45] Finished 'create-package-json' after 6.28 ms
[16:39:45] Finished 'build' after 29 s

sureshchahal committed:
Fixed complie errors

sureshchahal commented:
I have fixed those errors. Also provided AuthHttp class. This will acquire
token behind the scene and set header of request. It also work for external
API, if you specify those in config.

On Mon, Aug 22, 2016 at 2:06 AM, Neil665 notifications@github.com wrote:

> Hi sureshchahal, thanks for adding this functionality! (Thanks also to
> alenny for the original code!)
> 
> I wanted to try out your code so I cloned it from https://github.com/
> sureshchahal/angular2-adal and then I ran 'npm install' and 'gulp build'.
> I got errors because there were two TypeScript declaration files for
> adal.js, so I removed the line in typings.json so I was only left with the
> one under 'src\adal'. However there were still a number of build errors, as
> shown below. Were you using a more updated declaration file for adal.js?
> 
> [16:39:17] Using gulpfile ~\git\sureshchahal\angular2-adal\gulpfile.js
> [16:39:17] Starting 'build'...
> [16:39:17] Starting 'clean:dist'...
> [16:39:17] Finished 'clean:dist' after 9.92 ms
> [16:39:17] Starting 'readme:dist'...
> [16:39:17] Starting 'scripts:cjs'...
> [16:39:17] Starting 'scripts:ts'...
> [16:39:17] Finished 'readme:dist' after 260 ms
> src\services\adal.service.ts(37,16): error TS2339: Property
> 'AuthenticationContext' does not exist on type 'Window'.
> src\services\adal.service.ts(72,34): error TS2339: Property 'callback'
> does not exist on type 'AuthenticationContext'.
> src\services\adal.service.ts(72,59): error TS2339: Property
> 'callBackMappedToRenewStates' does not exist on type 'Window'.
> src\services\adal.service.ts(77,45): error TS2339: Property 'callback'
> does not exist on type 'AuthenticationContext'.
> src\services\adal.service.ts(84,46): error TS2339: Property 'callback'
> does not exist on type 'AuthenticationContext'.
> src\services\adal.service.ts(84,72): error TS2339: Property '_getItem'
> does not exist on type 'AuthenticationContext'.
> src\services\adal.service.ts(84,98): error TS2339: Property 'CONSTANTS'
> does not exist on type 'AuthenticationContext'.
> src\services\adal.service.ts(89,46): error TS2339: Property 'callback'
> does not exist on type 'AuthenticationContext'.
> src\services\adal.service.ts(89,72): error TS2339: Property '_getItem'
> does not exist on type 'AuthenticationContext'.
> src\services\adal.service.ts(89,98): error TS2339: Property 'CONSTANTS'
> does not exist on type 'AuthenticationContext'.
> src\services\adal.service.ts(90,46): error TS2339: Property
> '_renewFailed' does not exist on type 'AuthenticationContext'.
> src\services\authHttp.service.ts(2,39): error TS2307: Cannot find module
> '@angular/http'.
> src\services\authHttp.service.ts(3,27): error TS2307: Cannot find module
> 'angular2-adal/core'.
> [16:39:21] TypeScript: 13 semantic errors
> [16:39:21] TypeScript: emit succeeded (with errors)
> [16:39:21] Finished 'scripts:ts' after 4.63 s
> [16:39:21] Finished 'scripts:cjs' after 4.72 s
> [16:39:21] Starting 'scripts'...
> [16:39:21] Finished 'scripts' after 4.03 μs
> [16:39:21] Starting 'bundle:cjs'...
> [16:39:23] Starting 'bundle:cjs:min'...
> [16:39:45] Finished 'bundle:cjs' after 24 s
> [16:39:45] Finished 'bundle:cjs:min' after 23 s
> [16:39:45] Starting 'bundle'...
> [16:39:45] Finished 'bundle' after 5.87 μs
> [16:39:45] Starting 'copy-typings'...
> [16:39:45] Finished 'copy-typings' after 2.92 ms
> [16:39:45] Starting 'create-package-json'...
> [16:39:45] Finished 'create-package-json' after 6.28 ms
> [16:39:45] Finished 'build' after 29 s
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/alenny/angular2-adal/pull/11#issuecomment-241330301,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ARLaH6RyNkwPkE5BtRUUo8ScOvFVexvbks5qiUqAgaJpZM4JZ97p
> .

Neil665 commented:
Thanks, I've tried it out and it is working well.

@alenny are you thinking of merging this pull request into the main library?

jdkb123 committed:
AuthHttp can take RequestOptions in http requests

AuthHttp takes RequestOptions in get, post, delete just like underlying Http from angular. Makes a copy of the RequestOptions before passing to Http.

sureshchahal committed:
Merge pull request #1 from jdkb123/patch-1

AuthHttp can take RequestOptions in http requests

sureshchahal committed:
Update README.md

sureshchahal committed:
Only check for authenticated user only if backend API is configured in adal config

Also fix error if server return 204 status

kaleemxii commented on src/services/adal.service.ts:
```diff
@@ -83,7 +112,7 @@ export class AdalService {
+        })();
     }
 
     public getUser(): Observable<adal.User> {
```
can you update the below line 
        return Observable.bindCallback(function (cb: (u: adal.User) => void) { 
to 
       return Observable.bindCallback((cb: (u: adal.User) => void) => {

as  the ts compiler is behaving weirdly with first format and creating a JS of 

AdalService.prototype.getUser = function () {
        return Observable_1.Observable.bindCallback(function (cb) {
            this.adalContext.getUser(function (error, user) {

instead of this

```
AdalService.prototype.getUser = function () {
    var _this = this;
    return Observable_1.Observable.bindCallback(function (cb) {
        _this.adalContext.getUser(function (error, user) {
```

which is failing with null reference at this.adalContext.something

sureshchahal replied:
Updated

kaleemxii commented on src/services/authHttp.service.ts:
```diff
@@ -0,0 +1,100 @@
+    ,private adalService : AdalService
+    ) { }
+
+    get(url: string, options?: RequestOptionsArgs): Observable<any> {
```
if you are making this , it would be really helpful if you can include these methods as well

PATCH
PUT
HEAD

sureshchahal replied:
done

kaleemxii commented on src/services/authHttp.service.ts:
```diff
@@ -0,0 +1,100 @@
+        }
+
+        var resource = this.adalService.GetResourceForEndpoint(url);
+        var authenticatedCall = this.http.request(url, options).map(this.extractData).catch(this.handleError);
```
I would rather have this assigning statement only if resource is undefined, rather than overriding it.

sureshchahal replied:
done

kaleemxii reviewed (commented):
Thanks a lot for this PR Suresh, I was also making something similar to get the acquire token working , and then your PR saved the day..

i hope you get these changes published soon..

kaleemxii commented:
@alenny Any plans on merging this and releasing , Eagerly waiting for the release. 
Thanks in Advance

sureshchahal commented:
Hello kaleemxii- if you can make these changes and submit a change request
I will be happy to merge these. In case you are unable to make these
changes.. I will be working tonight on this

On Thursday, September 29, 2016, kaleemxii notifications@github.com wrote:

> @alenny https://github.com/alenny Any plans on merging this and
> releasing , Eagerly waiting for the release.
> Thanks in Advance
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/alenny/angular2-adal/pull/11#issuecomment-250575441,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ARLaH6VLEUuWTh3jyGwI9BdFi19-fdelks5qvBmKgaJpZM4JZ97p
> .

sureshchahal committed:
Provided Put, Patch and Head methods

sureshchahal commented:
made the changes. please take latest

On Thu, Sep 29, 2016 at 4:08 PM, suresh chahal sureshchahal@gmail.com
wrote:

> Hello kaleemxii- if you can make these changes and submit a change request
> I will be happy to merge these. In case you are unable to make these
> changes.. I will be working tonight on this
> 
> On Thursday, September 29, 2016, kaleemxii notifications@github.com
> wrote:
> 
> > @alenny https://github.com/alenny Any plans on merging this and
> > releasing , Eagerly waiting for the release.
> > Thanks in Advance
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > https://github.com/alenny/angular2-adal/pull/11#issuecomment-250575441,
> > or mute the thread
> > https://github.com/notifications/unsubscribe-auth/ARLaH6VLEUuWTh3jyGwI9BdFi19-fdelks5qvBmKgaJpZM4JZ97p
> > .

kaleemxii commented:
Wow that was quick, appreciate it :+1:

sureshchahal committed:
Released as separate package ng2-adal

sureshchahal commented:
published on NPM: https://www.npmjs.com/package/ng2-adal

On Thu, Sep 29, 2016 at 4:56 PM, kaleemxii notifications@github.com wrote:

> Wow that was quick, appreciate it 👍
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/alenny/angular2-adal/pull/11#issuecomment-250588921,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ARLaH5SX6EX6LaZ0E367iHxeT16Vl44Wks5qvCYSgaJpZM4JZ97p
> .

ericivarsson commented:
Thanks @sureshchahal, great work. Thanks also to @alenny for the original code.

fume committed:
body was ignored in Put, Post and Patch

sureshchahal committed:
Merge pull request #2 from fume/master

body was ignored in Put, Post and Patch

sureshchahal committed:
upgraded to angular2

sureshchahal committed:
Merge branch 'master' of https://github.com/sureshchahal/angular2-adal

sureshchahal committed:
upgraded version for publish

sureshchahal committed:
Fixed for Angular2

fume committed:
removed debugger

sureshchahal committed:
Merge pull request #7 from fume/issue-6

removed debugger

fume committed:
exposing via NgModule

fume committed:
Added skipLibCheck to tsconfig

fume committed:
Merge branch 'master' into issue-8

sureshchahal committed:
Merge pull request #9 from fume/issue-8

Expose as NgModule

sureshchahal committed:
Upgraded version after modules

sureshchahal committed:
Upgraded to 0.2.5

pkalinnikov committed:
Typo in imports. adal.service and authHttp.service do not work on Ubuntu

sureshchahal committed:
Merge pull request #14 from pkalinnikov/master

Typo in imports. adal.service and authHttp.service do not work on Ubuntu

Commit:
Renamed adal to adal-angular to let webpack build run without aliases

Commit:
Fixed webpack warnings "There is another module with an equal name when case is ignored."

Commit:
Updated gulp-typescript to use TypeScript 2.x and fix syntax errors during the build

sureshchahal committed:
Merge pull request #16 from DarkXaHTeP/master

Added ability to use with webpack

Commit:
Replaced typings with predefined typescript types for es2015 and browser DOM

Commit:
Updated angular-adal

Commit:
Removed postinstall for typings

Commit:
Removed parts of adal typing in src/adal-angular that are not needed after addition of @types for adal

sureshchahal committed:
Merge pull request #18 from DarkXaHTeP/master

Replacing typings with @types

Commit:
Updated version to 0.2.6

sureshchahal committed:
Update README.md

Commit:
fixing issue #19 and #20

cbrianball committed:
Fixed issue with 'this' not referencing AdalService as expected. Also updated Observable creation to proprly signify error condition.

sureshchahal committed:
Merge pull request #27 from cbrianball/bug/AdalService.acquireToken-onerror

Fixed issue with 'this' not referencing AdalService as expected. Also…

corfor committed:
acquireToken(resource) now returns Observable<string>, so Typescript can compile.

corfor committed:
fixed sendRequest(url, options) to return Observable<string>

corfor committed:
fixed runtime bug found by @jbarr33

corfor committed:
Merge branch 'develop'

corfor committed:
fixed runtime bug found by @jbarr33

mkelan committed:
Expose ability to refresh the data from cache

sureshchahal committed:
Merge pull request #30 from corfor/master

acquireToken(resource) now returns Observable<string>

Commit:
Merged and published latest pull

sureshchahal committed:
Merge pull request #35 from mazhisai/master

Expose ability to refresh the data from cache

Commit:
Updated version after publishing 0.3.0 version to npm

Cyberdada committed:
Update authHttp.service.ts

Changed line #64 From 
options1.headers.append('Authorization', 'Bearer ' + token);
options1.headers.set('Authorization', 'Bearer ' + token);

If you do several calls to the same service repeatedly you would get an error because the call appended several Bearer ... lines 
also cleaned up rxjs call according to requests

Cyberdada committed:
changed import from rxjs

sureshchahal committed:
Merge pull request #39 from Cyberdada/master

Changed import of Observable according to requests, and changed the way auth header is added.

jjbrunton committed:
Added check for id_token to fix the issue where token refreshes in the iframe didn't inform the parent that a token had been retrieved

cargowire committed:
Merge pull request #1 from Dootrix/bug/token-refresh

Added check for id_token to fix the issue where token refreshes in th…

sureshchahal committed:
Merge pull request #45 from Dootrix/master

Fix for refresh ignoring id_token

Commit:
Reverted pull from Cyberdada , it broke build

Cyberdada committed:
Update adal.service.ts

added import of bindCallback, and did some minor linting changes

Cyberdada committed:
Update authHttp.service.ts

Added necessary imports for rxjs and did some minor linting

Cyberdada committed:
Update core.ts

made export explicit

Cyberdada committed:
Update package.json

upgraded to Angular 4.1, and Typescript 2.3.3

Cyberdada committed:
Update bundle.js

Updated so that bundle points to correct angular files

Cyberdada committed:
Update authHttp.service.ts

removed explicit string declaration

sureshchahal committed:
Merge branch 'master' into master

sureshchahal committed:
Merge pull request #48 from Cyberdada/master

Updated to angular 4.1 and added the necessary rxjs imports so that code does not break.

sureshchahal committed:
Separate repo for ng2 and ng4

Updated description to mention separate repo for angular 2 and angular 4

Commit:
Published Angular 4 version

ericivarsson committed:
Don´t log errors to console

I might be wrong but I think it would be better to just return the error and leave all error handling to the client/initiator.

I have a problem right now where I´m calling the User/Photos endpoint of Microsoft Graph API. If a user doesn´t have a photo, the API return an 404 by design (https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/api/profilephoto_get). In this case I don´t want these 404´s logged to the console.

rjwijnen committed:
Merge pull request #1 from sureshchahal/master

Get latest

Commit:
Fix Throw error in AuthHttp.service

sureshchahal committed:
Merge pull request #53 from ericivarsson/patch-1

Don´t log errors to console

sureshchahal committed:
Merge pull request #54 from rjwijnen/master

Fix Throw error AuthHttp.service

aciccarello committed:
Fix reference to this in getUser error

Closes #24

aciccarello committed:
Eliminate dead code in adal.service

The user object is always being defaulted so if (user) will always be true.

nelsonmorais committed:
Fix Typescript >= 2.4.1 compilation error described on issue #64

For a full description of the problem and suggested fix see https://github.com/sureshchahal/angular2-adal/issues/64

sureshchahal committed:
Merge pull request #61 from aciccarello/patch-3

Fix reference to this in getUser error

sureshchahal committed:
Merge pull request #63 from aciccarello/patch-4

Eliminate dead code in adal.service

sureshchahal committed:
Merge branch 'master' into patch-1

sureshchahal committed:
Merge pull request #65 from nelsonmorais/patch-1

Fix Typescript >= 2.4.1 compilation error described on issue #64

Commit:
Updated package version

sureshchahal committed:
Update README.md

Read me for angular 4

sureshchahal committed:
Update README.md

sureshchahal committed:
Update package.json

bumped version to 2.0 to track angular 4 version.

Commit:
Update readme

Added the breaking change notice and improved formatting

sureshchahal committed:
Merge pull request #69 from claudiuconstantin/patch-1

Update readme

ranveeraggarwal committed:
Updated README with the new example URL

sureshchahal committed:
Merge pull request #70 from ranveeraggarwal/update-example-url

Updated README with the new example URL

rjwijnen committed:
Merge pull request #2 from sureshchahal/master

Merge

Commit:
Update package.json to angular 5

Commit:
Update add interceptor instead of AuthHttp

Commit:
Updated Readme

wvanderdeijl committed:
this was undefined in getUser error handler

you get here when calling getUser() before logging in. The adal library then throws 'User information is not available' string

Commit:
Add authenticationStatus observable to OAuthData

Commit:
Make authentication status observable on oauthData

Commit:
remove package lock

Commit:
Remove handle error since it is not doing anything

Commit:
Add observable.throw

sureshchahal committed:
Merge pull request #84 from straach/merge_back_to_fork_origin

Allow to retrieve login status on adalService asynchronously via observables

sureshchahal committed:
Merge pull request #80 from wvanderdeijl/getUser-this-undefined

this was undefined in getUser error handler

sureshchahal committed:
Merge pull request #77 from rjwijnen/master

Add angular 5 and replace AuthHttp with new Http Interceptor
